aequilibrae/paths/route_choice.py

```diff
@@ -0,0 +1,293 @@
+            until the number of desired routes is being consistently returned.
+
+        - When using LP, `max_depth` corresponds to the maximum number of iterations performed. While not enforced,
+            it should be higher than `max_routes`. It's value is dependent on the magnitude of the cost field,
+            specifically it's related to the log base `penalty` of the ratio of costs between two alternative routes.
+
+        :Arguments:
+            **par** (:obj:`dict`): Dictionary with all parameters for the chosen VDF
+        """
+
+        if any(key not in self.default_paramaters for key in par.keys()):
+            raise ValueError("Invalid parameter provided")
+
+        self.paramaters = self.default_paramaters | par
+
+    def set_save_path_files(self, save_it: bool) -> None:
+        """Turn path saving on or off.
+
+        :Arguments:
+            **save_it** (:obj:`bool`): Boolean to indicate whether paths should be saved
+        """
+        self.save_path_files = save_it
+        raise NotImplementedError()
+
+    def set_save_routes(self, where: Optional[str] = None) -> None:
+        """
+        Set save path for route choice resutls. Provide ``None`` to disable.
+
+        **warning** enabling route saving will disable in memory results. Viewing the results will read the results
+        from disk first.
+
+        :Arguments:
+            **save_it** (:obj:`bool`): Boolean to indicate whether routes should be saved
+        """
+        self.where = pathlib.Path(where) if where is not None else None
+
+    def prepare(self, nodes: Union[List[int], List[Tuple[int, int]]]):
+        """
+        Prepare OD pairs for batch computation.
+
+        :Arguments:
+            **nodes** (:obj:`Union[list[int], list[tuple[int, int]]]`): List of node IDs to operate on. If a 1D list is
+                provided, OD pairs are taken to be all pair permutations of the list. If a list of pairs is provided
+                OD pairs are taken as is. All node IDs must be present in the compressed graph. To make a node ID
+                always appear in the compressed graph add it as a centroid. Duplicates will be dropped on execution.
+        """
+        if len(nodes) == 0:
+            raise ValueError("`nodes` list-like empty.")
+
+        if isinstance(nodes[0], tuple):
+            # Selection of OD pairs
+            if any(len(x) != 2 for x in nodes):
+                raise ValueError("`nodes` list contains non-pair elements")
+            self.nodes = nodes
+
+        elif isinstance(nodes[0], int):
+            self.nodes = list(itertools.permutations(nodes, r=2))
+
+    def execute_single(self, origin: int, destination: int):
+        if self.__rc is None:
+            self.__rc = RouteChoiceSet(self.graph)
+
+        return self.__rc.run(origin, destination, **self.paramaters)
+
+    def execute(self, path_size_logit: bool = False):
```

Comment:
Let's change the name of the parameter to **perform_assignment** *=True*.

Eventually, we will have other route choice algorithms in there (e.g. remulaa), and then we add a method for *set_algorithm*